client/ayon_core/hosts/resolve/api/lib.py

```diff
@@ -409,6 +409,9 @@ def get_current_timeline_items(
         }
         # get track item object and its color
         for clip_index, ti in enumerate(_clips[track_index]):
+            if not ti.GetMediaPoolItem():
```

Comment:
Should we add a comment as to when this can occur?

For example:
```python

            # Skip items that do not have a media pool item, like for example
            # an "Adjustment Clip" or a "Fusion Composition" from the effects
            # toolbox
```

Also, is this the right place to fix this?

From the function name I'd expect this entry to be returned since it **is** a timeline item - it just doesn't have a mediapool item equivalent. I fixed it myself in `get_timeline_item` along the lines of:
```python
def get_timeline_item(media_pool_item: object,
                      timeline: object = None) -> object:
    """ ... """
    clip_name = media_pool_item.GetClipProperty("File Name")
    output_timeline_item = None
    timeline = timeline or get_current_timeline()

    with maintain_current_timeline(timeline):
        # search the timeline for the added clip

        for ti_data in get_current_timeline_items():
            ti_clip_item = ti_data["clip"]["item"]
            ti_media_pool_item = ti_clip_item.GetMediaPoolItem()

            # Skip items that do not have a media pool item, like for example
            # an "Adjustment Clip" or a "Fusion Composition" from the effects
            # toolbox
            if not ti_media_pool_item:
                continue

            if clip_name in ti_media_pool_item.GetClipProperty("File Name"):
                output_timeline_item = ti_clip_item

    return output_timeline_item
```

Since that particular function **does** try to get a matching timeline item to a media pool item.


Also, the logic in that function is much better of directly returning the clip in the if statement instead of storing it in a variable and continuing the loop. Since we'll always be returning one. If we want to preserve the current behavior where we return the LAST one found in the iterations we can do `for ti_data in reversed(get_current_timeline_items()):` just so that the first one we find is the last one returned by that other function.
I'm not sure however if the order things are being returned in there is in any particular order we care about or not?

Also, it seems pretty odd to match a timeline to a media pool item even though it may appear more than once on the timeline. I guess that's the current behavior and there's no bug _yet_ due to that? But it's definitely waiting for odd stuff to occur if we rely on there ever being only one entry present on the timeline of a media clip item. No?

Reply:
good point! thanks for suggestion. I had implemented it your way ;)